core/gui/src/app/workspace/service/report-generation/report-generation.service.ts

```diff
@@ -99,171 +105,195 @@ export class ReportGenerationService {
   }
 
   /**
-   * Retrieves and processes detailed results for a specified operator, generating a structured HTML representation.
-   * This function covers multiple cases, including handling paginated results, snapshot data, and scenarios where
-   * no results are found for the operator. The resulting HTML content is stored in the `allResults` array.
-   *
-   * @param {string} operatorId - The unique identifier of the operator for which results are being processed.
-   * @param {any} operatorInfo - Metadata and configuration details of the operator, utilized to render a JSON editor within the HTML.
-   * @param {Array<{operatorId: string, html: string}>} allResults - An array used to collect HTML content for each operator's processed result.
+   * Retrieves and processes results for all specified operators within the workflow.
+   * This function iterates over each operator ID, fetches the corresponding result details via `retrieveOperatorInfoReport`,
+   * and collects these results into an array. The function returns an observable that emits the processed results,
+   * which can be used to generate a comprehensive HTML report or for further processing.
    *
-   * @returns {Promise<void>} - A promise that resolves once the operator's results have been processed and the HTML has been appended to `allResults`.
+   * @param operatorId
+   * @param allResults
+   * @returns {Promise<void>}
    */
-  public retrieveOperatorInfoReport(
+  public async retrieveOperatorInfoReport(
     operatorId: string,
     allResults: { operatorId: string; html: string }[]
   ): Promise<void> {
```

Comment:
can you make it return a Observable<{operatorId: string; html: string}[]>? And remove async and Promise